__tests__/app.test.js

```diff
@@ -323,14 +323,71 @@ describe(`/api/articles`, () => {
         });
     });
 
-    test("response should be sorted by date in descending order", () => {
+    test("response should have default ordering by date and default sorting of descending", () => {
       return request(app)
         .get("/api/articles")
         .then(({ body: { articles } }) => {
           expect(articles).toBeSortedBy("created_at", { descending: true });
         });
     });
 
+    test("should accept query to sort by specific column", () => {
+      return request(app)
+        .get("/api/articles?sort=votes")
+        .then(({ body: { articles } }) => {
+          expect(articles).toBeSortedBy("votes", { descending: true });
+        });
+    });
+
+    test("should accept query to specify ordering type", () => {
+      return request(app)
+        .get("/api/articles?order=asc")
+        .then(({ body: { articles } }) => {
+          expect(articles).toBeSortedBy("created_at", { ascending: true });
+        });
+    });
+
+    test("should accept query to filter response by article topic", () => {
+      const query = "mitch";
+      return request(app)
+        .get(`/api/articles?topic=${query}`)
+        .then(({ body: { articles } }) => {
+          expect(articles).toBeSortedBy("created_at", { descending: true });
+          expect(articles).toEqual(
+            expect.arrayContaining([expect.objectContaining({ topic: query })])
```

Comment:
I'm not convinced by the assertion on line 357 - what would be a better assertion here is more of a check to see if every article you get back contains this topic. So I'd love to see a loop in this test block